volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -105,40 +105,62 @@ def get_sections(self):
             yield attr
 
     def get_symbols(self):
-        if symbols.symbol_table_is_64bit(self._context, self.get_symbol_table().name):
+        """Get module symbols"""
+        if symbols.symbol_table_is_64bit(self._context, self.get_symbol_table_name()):
             prefix = "Elf64_"
         else:
             prefix = "Elf32_"
         elf_table_name = intermed.IntermediateSymbolTable.create(
-            self.context,
-            self.config_path,
+            self._context,
+            self._context.modules["kernel"].config_path,
```

Comment:
This assumes that the plugin requested a module called `kernel`.  It's unclear why the config path has been changed, that's just a location to store configuration information.  Also, variables that start with an `_` are private, and `self.context` is just a read-only property that provides `self._context` so it's not clear why this change is necessary?  Please don't include changes like this without explaining why they're being made and why they're necessary...

Reply:
Hello, I am trying to get more familiar with the code base, and even if it effectively works locally for me, it may not be the best in terms of context. I'll be happy to work this out to make it cleaner.


So here basically, the two objects I replaced wheren't accessible anymore from the "self" object, which is why I had to resort to this. One solution would be to make them to the "module" object from the class instantiating.

Reply:
If config_path is indeed an arbitrary string ("kdbg" seems a convenient choice given the function you provided), I assume naming it `module` in the current context should be right ? 

However, if by "conflict" you mean that the config_path have to be unique for each instantiated module, we could try to concatenate the module address to the "module" string to be sure it is unique.

Reply:
`module` would work, but rather than reinstantiating the table for each time this method is called (and potentially making lots and lots of tables), it would be better to give it a known name (long and descriptive).  Ideally you could make the plugin load up the symbols it wants and then just pass them in as a table name, and thus remove the need for this bit of code to decide?